### ISSUE TITLE
MeanTimeReporter stores stats globally

### ISSUE THREAD

**Issue by tonydewan** [minitest-reporters/minitest-reporters#320]
That means that if you're using MeanTimeReporter on multiple projects on the same computer, the stats get mixed.

This is easy enough to override by setting the options, e.g.:

```
Minitest::Reporters::MeanTimeReporter.new(
  previous_runs_filename: Rails.root.join("tmp/minitest_reporters_previous_run"),
  report_filename: Rails.root.join("tmp/minitest_reporters_report")
)
```

But it seems like this should be the default. Or maybe I'm missing something?

I'm happy to make a PR for this if I've understood things correctly and the maintainers think it would be desirable to add.

**Comment by itsliamegan** [comment]
I agree this is an issue, but I'm not immediately sure of a good fix. Currently the report file is stored just in the system's temporary file directory. To be project-specific, it would make sense to store it in a project-specific temporary folder (like Rails' tmp/) but not every project has one of those.

**Comment by tonydewan** [comment]
I don't have an obvious bulletproof solution, though I had one idea that might work: hash the relevant caller path in the initializer and use that to make a directory or filename for the given project. This is mostly off the top of my head, but maybe something like:

```ruby
      def initialize(options = {})
        tmp_dir = Digest.hexencode(caller.detect{|c| c.include?("`new'")})
        ...
      end
```

One obvious problem with this is that changes to the caller location would lose history. I'm not sure how common that would be, thought?

**Comment by os97673** [comment]
Why not just provide a way to specify the location in command line? IMHO it will be hard to find the “default” witch will work for all.

**Comment by itsliamegan** [comment]
I think the issue with leaving it up to configuration is that the behavior is broken by default.
